src/main/java/org/jabref/gui/preferences/PreferencesSearchHandler.java

```diff
@@ -0,0 +1,140 @@
+            clearHighlights();
+            if (newSearchText.isEmpty()) {
+                clearSearch();
+            } else {
+                updateSearch(newSearchText);
+            }
+        });
+    }
+
+    private void updateSearch(String newSearchText) {
+        filterByTabName(newSearchText);
+        filterByLabelName(newSearchText);
+    }
+
+    private void filterByTabName(String newSearchText) {
+        List<PrefsTab> filteredTabs = preferenceTabs.stream()
+                                                    .filter(tab -> tab.getTabName().toLowerCase(Locale.ROOT).contains(newSearchText))
+                                                    .collect(Collectors.toCollection(ArrayList::new));
+        filteredPreferenceTabs.setAll(filteredTabs);
+    }
+
+    private void filterByLabelName(String newSearchText) {
+        for (PrefsTab tab : preferenceTabsLabelNames.keySet()) {
+            // If the current tab contains a matching label
+            boolean tabContainsLabel = false;
+            for (LabeledWrapper labeledWrapper : preferenceTabsLabelNames.get(tab)) {
+                if (labeledWrapper.contains(newSearchText)) {
+                    tabContainsLabel = true;
+                    highlightLabel(labeledWrapper);
+                }
+            }
+
+            if (tabContainsLabel) {
+                filteredPreferenceTabs.add(tab);
+            }
+        }
+    }
+
+    private void highlightLabel(LabeledWrapper labeledWrapper) {
+        labeledWrapper.setHighlighted(true);
+        highlightedLabels.add(labeledWrapper);
+    }
+
+    private void clearHighlights() {
+        highlightedLabels.forEach(labeledWrapper -> labeledWrapper.setHighlighted(false));
+    }
+
+    private void clearSearch() {
+        filteredPreferenceTabs.setAll(preferenceTabs);
+    }
+
+    /*
+     * Traverse all nodes of a PrefsTab and return a
+     * mapping from PrefsTab to all its Labeled type nodes.
+     */
+    private ArrayListMultimap<PrefsTab, LabeledWrapper> getPrefsTabLabelMap() {
+        ArrayListMultimap<PrefsTab, LabeledWrapper> prefsTabLabelMap = ArrayListMultimap.create();
+        for (PrefsTab prefsTab : preferenceTabs) {
+            Node builder = prefsTab.getBuilder();
+            if (builder instanceof Parent) {
+                Parent parentBuilder = (Parent) builder;
+                for (Node child : parentBuilder.getChildrenUnmodifiable()) {
+                    if (child instanceof Labeled) {
+                        LabeledWrapper labeledWrapper = new LabeledWrapper((Labeled) child);
+                        if (!labeledWrapper.getText().isEmpty()) {
+                            prefsTabLabelMap.put(prefsTab, labeledWrapper);
+                        }
+                    }
+                }
+            }
+        }
+        return prefsTabLabelMap;
+    }
+
+    protected ListProperty<PrefsTab> getFilteredPreferenceTabsProperty() {
```

Comment:
Just the wording, remove the "get" from the name., e.g. just `FilteredPreviewPreferencesTabsProperty()` to conform to the javafx code naming conventions